### ISSUE TITLE
[no-unsafe-assignment] Unsafe Assignment on __dirname

### ISSUE THREAD

**Issue by RLashofRegas** [typescript-eslint/typescript-eslint#2209]
**Repro**

I followed the getting started guide and put the following in my `.eslintrc.js`:

```js
module.exports = {
    root: true,
    parser: '@typescript-eslint/parser',
    parserOptions: {
        tsconfigRootDir: __dirname,
        project: [ './tsconfig.eslint.json', './src/webspa/tsconfig.json' ]
    },
    plugins: [
        '@typescript-eslint',
    ],
    extends: [
        'eslint:recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:@typescript-eslint/recommended-requiring-type-checking'
    ],
    env: {
        "browser": true,
        "amd": true,
        "node": true
    }
};
```

**Expected Result**
No linting errors are flagged on __dirname because I set env.node=true.

**Actual Result**
Received `Unsafe assignment of any value.` on the __dirname assignment.

**Additional Info**
I know I must be doing something wrong, so this may be more of a question, but this seems like the best place to get help, thanks :)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.2.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `14.4.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
Could you provide some example code? "on the __dirname assignment" doesn't describe what you're doing exactly.

**Comment by RLashofRegas** [comment]
@bradzacher the linting error is occurring on my `.eslintrc.js` file in the line `tsconfigRootDir: __dirname`. Presumably the parser is not understanding that __dirname is a string? I would have thought that this would be handled by `env.node=true`.

**Comment by bradzacher** [comment]
> Presumably the parser is not understanding that __dirname is a string? I would have thought that this would be handled by env.node=true.

Nope. The parser is based off of typescript, and typescript does not care about eslint configuration.

Even if it did though - eslint's globals setting only declares that a variable is _defined_, not what its type is. This is because eslint does not deal with types.

Side note - even the parser doesn't care about anything in the eslint configuration! That is, aside from the `parserOptions` because that's all ESLint passes to it.

options for fixing this:
- use `@types/node`
- create a local type definition file declaring it
- don't lint your `.eslintrc.js` with type aware linting

**Comment by RLashofRegas** [comment]
Thanks @bradzacher for the detailed explanation, this helps a lot with my understanding. And helps me realize that I think the problem is that I am linting from the root of my project folder which is not where the main project I am trying to lint is (my project uses an asp.net core back-end with angular front end). In order to make this work I had to install eslint and it's dependencies in a separate node_modules folder in the root of my repo in addition to the node_modules in my angular project. I didn't install `@types/node` in the root so I'm sure that's the problem, but it brings me back to wondering if there's a better way to handle the eslint dependencies given that I want my `.eslintrc.js` config in the root of my project but my `node_modules` is in a sub-dir. I think I had originally run into problems with this because I use vscode-eslint and I saw [this comment](https://github.com/microsoft/vscode-eslint/issues/32#issuecomment-224839077) about the eslint npm module dirs being static. Looking a little deeper now it seems like I may be able to use the `eslint.workingDirectories` setting for that extension to fix this so I'll look into trying that later today. Any other insight or ideas you have around this would be greatly appreciated though :)

**Comment by bradzacher** [comment]
You can always just ignore your `.eslintrc.js`.
https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories

You won't get much out of linting it anyway.

**Comment by RLashofRegas** [comment]
Yea...I guess I'm just being difficult trying at all costs not to add anything to ignore unless I absolutely have to XD

Thanks for all the help, closing this.
